fili-generic-example/src/main/java/com/yahoo/wiki/webservice/data/config/auto/DruidNavigator.java

```diff
@@ -217,5 +208,13 @@ private void queryDruid(SuccessCallback successCallback, String url) {
                 },
                 url
         );
+
+        try {
+            //calling get so we wait until responses are loaded before returning and processing continues
+            responseFuture.get();
+        } catch (InterruptedException | ExecutionException e) {
```

Comment:
Yes, it wouldn't hurt to put a timeout there, just in case the system takes a long time for some reason. A timeout of 30 seconds should be sufficient, but could make it configurable if you really wanted.

In general, timeouts and bounded buffers are good for keeping a system resilient 😉